### PR TITLE
refactor: Add more clear and tested fare contract availability status

### PR DESCRIPTION
**Pull request by gorandalum**
The logic whether a fare contract is upcoming, valid, expired, etc., 
was a little all over the place and hard to really get a grasp on. This
PR is made as an effort to be more precise on what these terms
actually are, and to make it easier to retrieve fare contracts with
specific availability status.

The new utils and hooks are not taken in use in this PR. I found
that taking them in use gave a lot of changes, but I wanted the PR
to be standalone to make possible for better reviews of the actual
logic, which is important to be correct and understandable. After
this is merged it will be another PR to replace old util functions
with usage of these new ones.

This will make the code inline with the terminology we have
described for fare contracts:
https://github.com/AtB-AS/docs-private/blob/main/terminology.md

### CODE REVIEW

**Review by rosvik — COMMENTED** [review]
Some feedback on the terminology: https://github.com/AtB-AS/docs-private/pull/67#issuecomment-2536307974

**Comment by gorandalum** [comment]
Updated the PR after the latest [terminology discussion](https://github.com/AtB-AS/docs-private/pull/67).

File: src/ticketing/utils.ts [hunk 1]
```diff
@@ -47,8 +47,11 @@ export function hasValidRightNowTravelRight(
   );
 }
 
-function isValidRightNowTravelRight(travelRight: TravelRight, now: number) {
-  // Unknown travel rights are not active
+export function isValidRightNowTravelRight(
```
Comment (rosvik) [inline src/ticketing/utils.ts:50]: I guess we don't need the "RightNow" parts of names anymore, if "valid" is always relative to `now`

**Review by rosvik — APPROVED** [review]
Great! Makes sense to me 💯

File: src/ticketing/utils.ts (hunk 1 above)
Comment (gorandalum) [reply src/ticketing/utils.ts]: This wasn't ment to be included for now, so I removed it. But you are right, we don't need to "RightNow" part, and hopefully we don't need this function at all after we start using `getAvailabilityStatus` around the app.

**Pull request merged**